src/qurveros/spacecurve.py

```diff
@@ -257,9 +257,23 @@ def set_params(self, params):
         quantities, the robustness properties and the control fields are
         evaluated for the new set of auxiliary parameters.
         """
-
+    def cast_leaf(x):
```

Comment:
It looks like the `cast_leaf` function has just been inserted where `set_params` was- this breaks `set_params`, and also doesn't look like it's actually updating `self.params` on line 276 anymore.